src/dos/drive_local.cpp

```diff
@@ -560,6 +566,36 @@ bool localFile::Close() {
 		fhandle = 0;
 		open = false;
 	};
+
+	if (newtime) {
+		// backport from DOS_PackDate() and DOS_PackTime()
+		struct tm tim = { 0 };
+		tim.tm_sec  = (time&0x1f)*2;
+		tim.tm_min  = (time>>5)&0x3f;
+		tim.tm_hour = (time>>11)&0x1f;
+		tim.tm_mday = date&0x1f;
+		tim.tm_mon  = ((date>>5)&0x0f)-1;
+		tim.tm_year = (date>>9)+1980-1900;
+		//  have the C run-time library code compute whether standard time or daylight saving time is in effect.
+		tim.tm_isdst = -1;
+		// serialize time
+		mktime(&tim);
+
+		struct utimbuf ftim;
+		ftim.actime = ftim.modtime = mktime(&tim);
+
+		char fullname[CROSS_LEN];
+		strcpy(fullname, basedir);
+		strcat(fullname, name);
```

Comment:
```suggestion
		char fullname[CROSS_LEN];
		snprintf(fullname, sizeof(fullname), "%s%s", basedir, name);
```
This way:
1. The code is protected from 2 buffer overflows
2. `fullname` buffer is guaranteed to be deliminated with 0-byte
3. MSVC won't freak out about with safety warnings around strcpy, strcat
4. Coverity won't freak out (same reasons)
5. And it's 1 line less and strcat won't need to calculate length of fullname for appending string ;)